# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/HostWebGLContext.cpp
# Commit: b9f43d8a6c2d
# Full Hash: b9f43d8a6c2d9504dde533d3db13111ae4d31d06
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-07-12 21:56:41
# Regressor Bug: 1777426
# File Overlap Count: 2
# Description:
#   Bug 1777426 - Support fast readbacks in CopyToSwapChain with async present. r=sotaro,jgilbert
#   
#   Currently CopyToSwapChain creates spurious copies of the back buffer when SharedSurfaces aren't exportable (= ToSurfaceDescriptor returns Nothing from SharedSurface_Basic). These then later get read back into a CPU memory buffer when PresentFrontBufferToCompositor is used to send the buffer to RemoteTextureMap. This has associated performance and memory costs.
#   
#   Conceptually, we want Present/CopyToSwapChain to just do the right thing and automatically push buffers to RemoteTextureMap, rather than secondarily needing a hidden call to PresentFrontBufferToCompositor. Then we can get rid of the need to create front buffers whose only purpose is to shuttle data to PresentFrontBufferToCompositor which then shuttles it RemoteTextureMap.
# ==============================================================================

diff -r 0a598b1533e6 -r b9f43d8a6c2d dom/canvas/HostWebGLContext.cpp
--- a/dom/canvas/HostWebGLContext.cpp	Tue Jul 12 06:49:11 2022 +0000
+++ b/dom/canvas/HostWebGLContext.cpp	Tue Jul 12 06:56:19 2022 +0000
@@ -91,11 +91,6 @@
   (void)mOwnerData.outOfProcess->SendJsWarning(text);
 }
 
-gl::SwapChain* HostWebGLContext::GetSwapChain(const ObjectId xrFb,
-                                              const bool webvr) const {
-  return mContext->GetSwapChain(AutoResolve(xrFb), webvr);
-}
-
 Maybe<layers::SurfaceDescriptor> HostWebGLContext::GetFrontBuffer(
     const ObjectId xrFb, const bool webvr) const {
   return mContext->GetFrontBuffer(AutoResolve(xrFb), webvr);